Sources/Bond/Collections/ObservableArray.swift

```diff
@@ -295,30 +296,30 @@ extension MutableObservableArray {
 }
 
 extension MutableObservableArray where Item: Equatable {
-  
+
   public func replace(with array: [Item], performDiff: Bool) {
-    if performDiff {
-      lock.lock()
+    lock.lock(); defer { lock.unlock() }
 
+    if performDiff {
       let diff = self.array.extendedDiff(array)
+      let patch = diff.patch(from: self.array, to: array)
```

Comment:
Using `ExtendedPatch`, my moves and reorders of **multiple items** work consistently. Using `ExtendedDiff`, the ordering is out of sequence (it looks like it's using the original indexes for elements in the array).  Moves of a single element work fine with both.

I am beginning to _suspect_ that this may be a difference between UIKit (UITableView) and AppKit (NSTableView).

I need to read the docs on how UITableView treats the indices of moved elements in between a `UITableView.beginUpdates()` and `UITableView.endUpdates()`, and whether it expects you to adjust the "from" index at each step, or just use the index of the item in the original array.